integ-tests/src/cucumber/java/steps/dataservice/DescribeEnvironmentSteps.java

```diff
@@ -0,0 +1,203 @@
+  private static final String CLUSTER_NAME_PREFIX = "cluster";
+  private String postfix = null;
+
+  public DescribeEnvironmentSteps() {
+    Given(
+        "^I create an environment named \"([^\"]*)\"$",
+        (final String environmentNamePrefix) -> {
+          this.postfix = generatePostfix();
+          dataServiceWrapper.createEnvironment(createEnvironmentRequest(environmentNamePrefix));
+        });
+
+    When(
+        "^I describe the environment named \"([^\"]*)\"$",
+        (final String environmentNamePrefix) -> {
+          dataServiceWrapper.describeEnvironment(describeEnvironmentRequest(environmentNamePrefix));
+        });
+
+    Then(
+        "^the created and described environments match$",
+        () -> {
+          final CreateEnvironmentResponse createEnvironmentResponse =
+              dataServiceWrapper.getLastFromHistory(CreateEnvironmentResponse.class);
+          final DescribeEnvironmentResponse describeEnvironmentResponse =
+              dataServiceWrapper.getLastFromHistory(DescribeEnvironmentResponse.class);
+          checkEnvironmentEquality(
+              createEnvironmentResponse.getEnvironment(),
+              describeEnvironmentResponse.getEnvironment());
+        });
+
+    Given(
+        "^I update the environment named \"([^\"]*)\"$",
+        (final String environmentNamePrefix) -> {
+          dataServiceWrapper.updateEnvironment(updateEnvironmentRequest(environmentNamePrefix));
+        });
+
+    Then(
+        "^the updated and described environments match$",
+        () -> {
+          final UpdateEnvironmentResponse updateEnvironmentResponse =
+              dataServiceWrapper.getLastFromHistory(UpdateEnvironmentResponse.class);
+          final DescribeEnvironmentResponse describeEnvironmentResponse =
+              dataServiceWrapper.getLastFromHistory(DescribeEnvironmentResponse.class);
+          checkEnvironmentEquality(
+              updateEnvironmentResponse.getEnvironment(),
+              describeEnvironmentResponse.getEnvironment());
+        });
+
+    When(
+        "^I try to describe a non-existent environment named \"([^\"]*)\"$",
+        (final String environmentNamePrefix) -> {
+          dataServiceWrapper.tryDescribeEnvironment(
+              describeEnvironmentRequest(environmentNamePrefix));
+        });
+
+    Given(
+        "^I delete the created environment named \"([^\"]*)\"$",
+        (final String environmentNamePrefix) -> {
+          dataServiceWrapper.tryDeleteEnvironment(deleteEnvironmentRequest(environmentNamePrefix));
+        });
+
+    When(
+        "^I try to describe the environment named \"([^\"]*)\"$",
+        (final String environmentNamePrefix) -> {
+          dataServiceWrapper.tryDescribeEnvironment(
+              describeEnvironmentRequest(environmentNamePrefix));
+        });
+  }
+
+  private DescribeEnvironmentRequest describeEnvironmentRequest(
```

Comment:
We can move the request wrappers into the dataservicewrapper class and reuse them from all steps

Reply:
Sure. I can move them into the wrappers.